src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -44,5 +43,47 @@ void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo)
 #endif
 }
 
+RefreshKeySender * DefaultCheckInDelegate::OnKeyRefreshNeeded(ICDClientInfo & clientInfo, ICDClientStorage * clientStorage)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    auto refreshKeySender = chip::Platform::New<RefreshKeySender>(this, clientInfo, clientStorage);
+    if (refreshKeySender == nullptr)
+    {
+        return nullptr;
+    }
+
+    err = chip::Crypto::DRBG_get_bytes(refreshKeySender->GetRefreshKeyBuffer().Bytes(),
+                                       refreshKeySender->GetRefreshKeyBuffer().Capacity());
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(ICD, "Generation of new key failed: %" CHIP_ERROR_FORMAT, err.Format());
+        Platform::Delete(refreshKeySender);
+        return nullptr;
+    }
+    mRefreshKeySenderMap[clientInfo.peer_node] = refreshKeySender;
```

Comment:
You don't need this mRefreshKeySenderMap at all, you just need pass the refreshKeySender pointer OnKeyRefreshDone, then it can take care of memory release for this pointer. thanks